gcore/gdalalgorithm.cpp

```diff
@@ -581,27 +582,133 @@ bool GDALAlgorithmArg::ValidateChoice(const std::string &value) const
 
 bool GDALAlgorithmArg::RunValidationActions()
 {
+    bool ret = true;
+
     if (GetType() == GAAT_STRING && !GetChoices().empty())
     {
         if (!ValidateChoice(Get<std::string>()))
-            return false;
+            ret = false;
     }
     else if (GetType() == GAAT_STRING_LIST && !GetChoices().empty())
     {
         const auto &values = Get<std::vector<std::string>>();
         for (const std::string &value : values)
         {
             if (!ValidateChoice(value))
-                return false;
+                ret = false;
         }
     }
 
+    const auto ValidateIntRange = [this, &ret](int val)
```

Comment:
Consider making this and `ValidateRealRange` member functions? Can `ValidateRealRange` be used for ints, since 64 bit ints are not handled?

Reply:
> Consider making this and ValidateRealRange member functions? 

done

> Can `ValidateRealRange` be used for ints, since 64 bit ints are not handled?

for the error messages, for the int case, I explicitly cast to int to get better output